### PR TITLE
[DOCS] Update dev mode tutorial

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
Update dev mode tutorial to give more realistic getting started steps, in this case, connecting to another application.

Closes https://github.com/ethereum/go-ethereum/issues/20397

### CODE REVIEW

File: docs/_getting-started/dev-mode.md [hunk 1]
```diff
@@ -12,13 +12,20 @@ Starting geth in dev mode does the following:
...
+```shell
+mkdir test-chain-dir
+```
+
+For this guide, start geth in dev mode, and enable [RPC](../_rpc/server.md) so you can connect other applications to geth. For this guide, we use Remix, the web-based Ethereum IDE, so also allow its domains to accept cross-origin requests.
 
 ```shell
-$ mkdir test-chain-dir
-$ geth --dev --datadir test-chain-dir console
+geth --datadir test-chain-dir  --port 8545 --rpc --rpcport "8545" --dev --rpccorsdomain "https://remix.ethereum.org,http://remix.ethereum.org" console
```
Comment (fjl) [inline docs/_getting-started/dev-mode.md]: `--port` is not needed here

Comment (fjl) [reply docs/_getting-started/dev-mode.md]: `--rpcport "8545"` doesnt' need the quotes, and is also not needed because that's the default port

**Comment by ChrisChinchilla** [comment]
Fixed up @fjl

File: docs/_getting-started/dev-mode.md [hunk 2]
```diff
@@ -41,8 +48,8 @@ And check the balance of the account:
...
 
-If you want to test your dapps with a realistic block time use the `--dev.period` option when you start dev mode:
+If you want to test your dapps with a realistic block time use the `--dev.period` option when you start dev mode with the `--dev.period 14` argument.
 
-```shell
-geth --dev --dev.period 14 console
-```
+## Connect Remix to Geth
+
+With geth now running, open <http://remix.ethereum.org> (Yes, use `http` when connecting to your local machine). Compile the contract as normal, but when you deploy and run a contract, select _Web3 Provider_ from the _Environment_ dropdown, and add "http://127.0.0.1:8545" to the popup box. Click _Deploy_, and interact with the contract. You should see contract creation, mining, and transaction activity.
```
Comment (fjl) [inline docs/_getting-started/dev-mode.md:55]: I'm not so happy to recommend 'http' over 'https' here. We need to make it work with the https version of remix.

Comment (ChrisChinchilla) [reply docs/_getting-started/dev-mode.md:55]: @fjl OK, so for a getting started guide, would a simple solution be using something like ngrok or localtunnel? Or do you know a better way to setup secure communication between a website and a local machine, within the scope of a getting started guide?

Comment (fjl) [reply docs/_getting-started/dev-mode.md:55]: https remix does work with Chrome and local geth if you use `http://127.0.0.1:8584` as the provider URL. It does not work if you use `localhost` in the provider URL, which is still the default in remix AFAIK.

Comment (fjl) [reply docs/_getting-started/dev-mode.md:55]: I wish we could find a way to make it work in Firefox as well.

Comment (holiman) [reply docs/_getting-started/dev-mode.md:55]: Can't we just merge this @fjl

**Comment by karalabe** [comment]
This PR seems good to us. @fjl Please comment if you think this should not be merged, otherwise we can push it in a couple days.

**Comment by ChrisChinchilla** [comment]
@holiman @fjl @karalabe I'll take a quick look at this today and see if I can find an "easy" solution to using https instead

**Comment by ChrisChinchilla** [comment]
Updated a few small things, but curiously this works in Chrome and FF now, using https, but yeah, I'll make a PR to Remix to adjust what they tell you to do by default.

**Pull request merged**